python/satyaml/TRISAT-R.yml

```
@@ -0,0 +1,46 @@
+name: TRISAT-R
+norad: 53108
+data:
+  &tlm unknown Telemetry:
+    trisat
```

I think there is a typo here. This should read
```suggestion
  &tlm Telemetry:
    telemetry: trisat
```
if the telemetry format of TRISAT-R is the same as TRISAT, or
```
  &tlm Telemetry:
    unknown
```
if not.

---

Indeed, my mistake. using `telemetry: trisat` gives the following output and seems to be correct.
To be save we can also use `unknown` I haven't found any TRISAT-R specific telemetry.

```
gr_satellites TRISAT-R.yml --rawint16 iq_48000_6658825.wav --samp_rate 48e3 --iq --use_agc
-> Packet from 1k baud FSK downlink
Container: 
    tm_primary_header = Container: 
        transfer_frame_version_number = 0
        spacecraft_id = 1
        virtual_channel_id = 1
        ocf_flag = True
        master_channel_frame_count = 81
        virtual_channel_frame_count = 12
        transfer_frame_secondary_header_flag = False
        synch_flag = True
        packet_order_flag = False
        segment_length_id = 0
        first_header_pointer = 211
    payload = b'K0 0\x11\x02\x00\x00\x00\x00\x15N\x00\x00\x01\x00\x00\x1a\x01\x00\x9a\x0b\xcatO\xc2\x00\x05\x00\x00\x08*(\xc1\xe2\\\xc1\x9d\x00\x00\x98U}\xb6\xbc\x94A\xb4\x00\x00\xae9\x96\x8b\x89\x1d\xea\x10\xfe\xff\x14\x15;\t\xf7\xf0\xf4\x94\x07\x00\xd1g\x0cm\xf6@\x91\xa1\x02\x00\xe5L\xff]\xae\x81:\x01\x00\x00E\x03\xcd\x1c\xc2\x02\xbb\x00\xde\x00\xbb\x7f\xe7\x07\xc7\x02\x00\x00\x00\x00\xdd[p\xff\xff\xff\xb2\xf6\xff\xff&Zi\xff\xff\xff\x00\x00\x00\x00\xa5\x16\xbd\xff\xff\xff\xa8\xf6\xff\xff\x94tw\xff\xff\xff\x00\x00\x00\x00T<}\x00\xac\xff<\x00\x00\x00X$y\x00\xaf\xff?\x00\x00\x00w>\xe1\xde\x1f!\x01\xc0\x00\x00aE\xc0\x05\x9e\x05=\x06\x00\x00aG9\xc1\x00\x00\x00\x00\x00\x00\xd4h\x00\x00\x00\x00\x00\x00\x00\x00\x95' (total 211)
    ocf = Container: 
        control_word_type = False
        clcw_version_number = 0
        status_field = 0
        cop_in_effect = 1
        virtual_channel_identification = 1
        rsvd_spare1 = 0
        no_rf_avail = True
        no_bit_lock = True
        lockout = False
        wait = False
        retransmit = False
        farmb_counter = 0
        rsvd_spare2 = False
        report_value = 1
    crc = 15279
```

Corrected: 

```
name: TRISAT-R
norad: 53108
data:
  &tlm Trisat Telemetry:
    telemetry: trisat
transmitters:
  1k baud FSK downlink:
    frequency: 435.6125e+6
    modulation: FSK
    baudrate: 1000
    framing: CCSDS Concatenated
    RS basis: dual
    convolutional: NASA-DSN
    frame size: 223
    data:
    - *tlm
  2k baud FSK downlink:
    frequency: 435.6125e+6
    modulation: FSK
    baudrate: 2000
    framing: CCSDS Concatenated
    RS basis: dual
    convolutional: NASA-DSN
    frame size: 223
    data:
    - *tlm
  3k baud FSK downlink:
    frequency: 435.6125e+6
    modulation: FSK
    baudrate: 3000
    framing: CCSDS Concatenated
    RS basis: dual
    convolutional: NASA-DSN
    frame size: 223
    data:
    - *tlm
  5k baud FSK downlink:
    frequency: 435.6125e+6
    modulation: FSK
    baudrate: 5000
    framing: CCSDS Concatenated
    RS basis: dual
    convolutional: NASA-DSN
    frame size: 223
    data:
    - *tlm
```

---

Do you have more packets in that recording? One way to check if the format is okay is to see if the `master_channel_frame_count` and `virtual_channel_frame_count` increase by one with each packet. So far some of the values look a bit suspicious, such as `first_header_pointer = 211`, so I'm more inclined to think that it doesn't use the same format as TRISAT.